pipelines/common/src/main/java/com/google/fhir/analytics/view/ViewApplicator.java

```diff
@@ -259,7 +260,18 @@ FlatRow applyColumns(@Nullable IBase element, List<Column> columns)
       if (element != null) {
         List<IBase> eval = fhirPath.evaluate(element, col.getPath(), IBase.class);
         // TODO fix this by handling types and avoiding `toString()`!
-        value = String.join(",", eval.stream().map(e -> e.toString()).collect(Collectors.toList()));
+        value =
+            String.join(
+                ",",
+                eval.stream()
+                    .map(
+                        e -> {
+                          if (e instanceof IPrimitiveType<?>) {
+                            return ((IPrimitiveType) e).getValueAsString();
```

Comment:
Thanks @chandrashekar-s for the review. FYI, I just added this change too which fixes an issue that I found when implementing other features. The test that is also added in this commit, would have failed without this change.

Reply:
Resolving this for now to merge; please let me know if you have any concerns and I will fix it in the follow-up PR that I am going to send shortly.